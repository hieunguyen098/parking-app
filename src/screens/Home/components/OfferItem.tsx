import { Image, StyleSheet, Text, View } from 'react-native';
import React from 'react';

const OfferItem = () => {
    return (
        <View style={styles.container}>
            <Image
                style={styles.image}
                source={{ uri: 'http://img.giftpop.vn/brand/GIFTPOP/MP1807250002_BASIC_origin.jpg' }}
            />
            <Text style={styles.text}>Ưu đãi 30.000 hấp dẫn!! Nhanh tay nhận voucher</Text>
        </View>
    );
};

export default OfferItem;

const styles = StyleSheet.create({
    container: { marginBottom: 16 },
    image: {
        width: '100%',
        height: 200,
        marginVertical: 8,
        borderRadius: 8,
    },
    text: {
        fontSize: 20,
        fontWeight: '400',
        color: '#404040',
    },
});
