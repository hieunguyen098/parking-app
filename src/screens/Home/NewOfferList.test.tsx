import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

import OfferItem from './components/OfferItem';
import { QueryClient, QueryClientProvider } from 'react-query';
import NewOffer from './components/NewOfferList';

jest.useFakeTimers();
test('test offer list', () => {
    const queryClient = new QueryClient();
    const component = (
        <QueryClientProvider client={queryClient}>
            <NavigationContainer>
                <NewOffer />
            </NavigationContainer>
        </QueryClientProvider>
    );

    const tree = render(component);
    expect(tree).toMatchSnapshot();
});
