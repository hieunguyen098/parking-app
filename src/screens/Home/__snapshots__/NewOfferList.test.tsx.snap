// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test offer list 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 24,
            "fontWeight": "700",
          },
          undefined,
        ]
      }
    >
      Ưu đãi mới
    </Text>
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FF953A",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Xem thêm
        </Text>
        <Text />
      </View>
    </View>
  </View>
  <Text>
    Loading
  </Text>
</View>
`;
